caikit_nlp/modules/text_generation/text_generation_local.py

```diff
@@ -590,7 +591,11 @@ def run_tokenizer(
             TokenizationResults
                 The token count
         """
-        raise NotImplementedError("Tokenization not implemented for local")
+        error.type_check("<NLP48137045E>", str, text=text)
+        tokenized_output = self.model.tokenizer(text)
```

Comment:
Very interesting question! HF in their [documentation](https://huggingface.co/docs/transformers/en/main_classes/tokenizer#transformers.PreTrainedTokenizer.__call__.return_attention_mask) mention that  _If left to the default, will return the attention mask according to the specific tokenizer’s default, defined by the return_outputs attribute._ 

For consistency and clarity with the other modules, it could be prudent to include `return_attention_mask` explicitly. In such a case, the only potential concern would be whether setting `return_attention_mask` explicitly could result in messing with the default behaviour of the tokeniser, but I think the likelihood of this is low. 

Thus, I can change the method to:
```
def run_tokenizer(
        self,
        text: str,
    ) -> TokenizationResults:
        """Run tokenization task against the model

        Args:
           text: str
                Text to tokenize
        Returns:
            TokenizationResults
                The token count
        """
        error.type_check("<NLP48137045E>", str, text=text)
        tokenized_output = self.model.tokenizer(text, return_attention_mask=True)
        return TokenizationResults(
            token_count=len(tokenized_output["input_ids"]),
        )
```

Let me know how you would like me to proceed

Reply:
I've added `return_attention_mask=True` based on our conversation to ensure consistency with the other modules